compiler/rustc_parse/src/lib.rs

```diff
@@ -481,7 +504,11 @@ pub fn tokentree_probably_equal_for_proc_macro(
             token_probably_equal_for_proc_macro(token, token2)
         }
         (TokenTree::Delimited(_, delim, tts), TokenTree::Delimited(_, delim2, tts2)) => {
-            delim == delim2 && tokenstream_probably_equal_for_proc_macro(&tts, &tts2, sess)
+            // `NoDelim` delimiters can appear in the captured tokenstream, but not
+            // in the reparsed tokenstream. Allow them to match with anything, so
+            // that we check if the two streams are structurally equivalent.
+            (delim == delim2 || *delim == DelimToken::NoDelim || *delim2 == DelimToken::NoDelim)
```

Comment:
Could this relaxed comparison be enabled by a flag rather than applied always?
It only makes sense when we are trying to reparse and compare for the second time.